KubeArmor/BPF/shared.h

```diff
@@ -24,6 +24,29 @@ char LICENSE[] SEC("license") = "Dual BSD/GPL";
 #define AUDIT_POSTURE 140
 #define BLOCK_POSTURE 141
 #define CAPABLE_KEY 200
+#define TTY_LEN 64
+
+#ifdef BTF_SUPPORTED
+#define GET_FIELD_ADDR(field) __builtin_preserve_access_index(&field)
+
+#define READ_KERN(ptr)                                    \
+    ({                                                    \
+        typeof(ptr) _val;                                 \
+        __builtin_memset((void *)&_val, 0, sizeof(_val)); \
+        bpf_core_read((void *)&_val, sizeof(_val), &ptr); \
+        _val;                                             \
+    })
+#else
+#define GET_FIELD_ADDR(field) &field
+
+#define READ_KERN(ptr)                                     \
+    ({                                                     \
+        typeof(ptr) _val;                                  \
+        __builtin_memset((void *)&_val, 0, sizeof(_val));  \
+        bpf_probe_read((void *)&_val, sizeof(_val), &ptr); \
+        _val;                                              \
+    })
+#endif
```

Comment:
no need to check for BTF support in case of bpf-lsm. it can be ignored either use READ_KERN macro with bpf_probe_read() def. or you can simply use bpf_probe_read() helper directly if it's not being used at other places both should be fine.